### ISSUE TITLE
Keptn 0.8.0-alpha CLI crashes for auth after upgrade from 0.7.3

### ISSUE THREAD

**Issue by GustavAT** [keptn/keptn#2912]
Upgrading the CLI from 0.7.3 to 0.8.0-alpha as described in the [Docs](https://keptn.sh/docs/0.8.x/operate/upgrade/) results in the CLI crashing when patching namespaces or authenticating again.

**Environment**

* Client OS (e.g., Linux, OSX, Windows): Ubuntu 20.04
* Keptn Version (`keptn version`): 0.8.0-alpha
* Kubernetes Cloud Provider (e.g., GKE, AKS): GKE
* Kubernetes version (`kubectl version`): v1.16.15-gke.4300

**Affected Component**

* [ ] Docs
* [X] CLI
* [ ] Kubernetes Integration
* [ ] Openshift Integration
* [ ] Helm
* [ ] Istio
* [ ] Bridge
* [ ] Gatekeeper
* [ ] Datastore
* [ ] REST API
* [ ] eventbroker / distributor
* [ ] jmeter

**Describe the bug**

After upgrading the CLI as described in the [Docs](https://keptn.sh/docs/0.8.x/operate/upgrade/), commands such as `keptn auth` or `keptn upgrade --patch-namespace` crash the CLI.

**To Reproduce**

Have the keptn CLI 0.7.3 installed.

Download the 0.8.0-alpha release (`keptn-0.8.0-alpha-linux-amd64.tar.gz
`)  from [GitHub](https://github.com/keptn/keptn/releases/tag/0.8.0-alpha) and extract it.

Replace old keptn installation:
`mv keptn-0.8.0-alpha-linux-amd64 /usr/local/bin/keptn`
Set execute permissions
`chmod +x /usr/local/bin/keptn`

`> keptn version` Shows the expected version and tells us that the CLI needs to be authenticated against the cluster
```bash
Keptn CLI version: 0.8.0-alpha
Keptn cluster version: This command requires to be authenticated. See "keptn auth" for details
```

`> keptn auth` crashes
```bash
panic: runtime error: index out of range [0] with length 0

goroutine 1 [running]:
github.com/keptn/keptn/cli/cmd.smartKeptnCLIAuth(0xc00003e0c5, 0xc00083dd58, 0xc00083dd58, 0x13de9a4)
	/home/runner/work/keptn/keptn/cli/cmd/auth.go:250 +0x617
github.com/keptn/keptn/cli/cmd.verifyAuthParams(0xc000211a70, 0x1, 0xc000522770)
	/home/runner/work/keptn/keptn/cli/cmd/auth.go:157 +0x455
github.com/keptn/keptn/cli/cmd.glob..func2(0x2accc40, 0x2b8a938, 0x0, 0x0, 0x0, 0x0)
	/home/runner/work/keptn/keptn/cli/cmd/auth.go:68 +0x72
github.com/spf13/cobra.(*Command).execute(0x2accc40, 0x2b8a938, 0x0, 0x0, 0x2accc40, 0x2b8a938)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:831 +0x514
github.com/spf13/cobra.(*Command).ExecuteC(0x2ad0600, 0xc0000b4058, 0x0, 0x0)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
github.com/keptn/keptn/cli/cmd.Execute()
	/home/runner/work/keptn/keptn/cli/cmd/root.go:56 +0x92
main.main()
	/home/runner/work/keptn/keptn/cli/main.go:34 +0x77

```

`> keptn auth --endpoint=$KEPTN_ENDPOINT --api-token=$KEPTN_API_TOKEN` works

It is mentioned in the docs, that namespaces should be patched after the upgrade using the following command:

`> keptn upgrade --patch-namespace` crashes (same error before and after CLI authentication)
```bash
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x172e2cd]

goroutine 1 [running]:
github.com/keptn/keptn/cli/cmd.getAppVersion(...)
	/home/runner/work/keptn/keptn/cli/cmd/upgrade.go:148
github.com/keptn/keptn/cli/cmd.doUpgrade(0x0, 0x0)
	/home/runner/work/keptn/keptn/cli/cmd/upgrade.go:194 +0x7d
github.com/keptn/keptn/cli/cmd.glob..func43(0x2ad2580, 0xc0008da1b0, 0x0, 0x1, 0x0, 0x0)
	/home/runner/work/keptn/keptn/cli/cmd/upgrade.go:65 +0xb3
github.com/spf13/cobra.(*Command).execute(0x2ad2580, 0xc0008da1a0, 0x1, 0x1, 0x2ad2580, 0xc0008da1a0)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0x2ad0600, 0xc0000b6058, 0x0, 0x0)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950 +0x349
github.com/spf13/cobra.(*Command).Execute(...)
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
github.com/keptn/keptn/cli/cmd.Execute()
	/home/runner/work/keptn/keptn/cli/cmd/root.go:56 +0x92
main.main()
	/home/runner/work/keptn/keptn/cli/main.go:34 +0x77

```

**Comment by christian-kreuzberger-dtx** [comment]
@ankitjain28may I'd like to assign this to you, as you seem to have fixed it in PR #2971. Please comment on this issue so I can assign it.

**Comment by ankitjain28may** [comment]
@christian-kreuzberger-dtx Please assign this to me :)
